### ISSUE TITLE
Support XWayland

### ISSUE THREAD

**Issue by bugaevc** [owl-compositor/owl#9]
There's some interest in using Owl + XWayland to potentially replace XQuartz. So we should look into being able to run XWayland, at the very least.

**Comment by RJVB** [comment]
Agreed, Apple's own Jeremy Huddleston has been telling me that Wayland would be the best way to get better X11 functionality on Mac (including support for compositing).

Don't Wayland and EGL go more or less hand-in-hand, and how do you solve that? Mesa can be built to provide the Gallium/lllvmpipe based EGL emulation but I understand that's only useable for (simple) UI effects...
